File: metalanguage/src/main/java/ports/MLPort.java
package ports;

import constructs.MLConstruct;
import lombok.Getter;
import lombok.NonNull;
import lombok.Setter;

import java.util.UUID;

@Getter
@Setter
public class MLPort extends MLConstruct {
    @NonNull
    private MLPortKinds kind;

    public MLPort(@NonNull String name, @NonNull UUID id, @NonNull UUID prototypeId, @NonNull MLPortKinds kind) {
        super(name, id, prototypeId);
        this.kind = kind;
    }
}

File: metalanguage-lib/src/main/kotlin/ru/psu/ports/MLPort.kt
package ru.psu.ports

import ru.psu.constructs.MLConstruct
import java.util.*

//Класс порта в MetaLanguage 1.1
class MLPort(
        parentId:UUID?, //Идентификатор конструкции-родителя
        id: UUID, //Id порта
        name: String, //Имя порта
        prototypeId: UUID?, //Id прототипа
        var kind: MLPortKinds //Вид порта
): MLConstruct(parentId, id, name, prototypeId, null)